sdk/python/kfp/v2/components/structures_test.py

```diff
@@ -21,20 +21,21 @@
 from absl.testing import parameterized
 from kfp.v2.components import structures
 
+
 V1_YAML_IF_PLACEHOLDER = textwrap.dedent("""\
     name: component_if
     inputs:
-    - {name: optional_input_1, type: String, optional: true}
```

Comment:
Any reason to change this? underscore is valid character that can be used in a Python identifier, while dash is not. 
dash has to be replaced to something else before user can use the component with a keyword argument.